apps/event-worker/src/job-sync/index.ts

```diff
@@ -47,9 +48,11 @@ export const createjobSyncWorker = () =>
           if (syncFunction == null) return;
 
           try {
-            syncFunction(je.job).then(
-              (isCompleted) => isCompleted && removeJobSyncJob(job),
-            );
+            syncFunction(je.job).then(async (isCompleted) => {
+              if (!isCompleted) return;
+              removeJobSyncJob(job);
+              await onJobCompletion(je.job);
```

Comment:
this should be awaited in case two are processed at the same time

Reply:
im confused could you explain this more